test/runtime_tests.cc

```diff
@@ -992,5 +992,39 @@ namespace detail {
 		});
 	}
 
+	void dry_run_with_nodes(const int nodes) {
+		const std::string dryrun_envvar_name = "CELERITY_DRYRUN_NODES";
+		setenv(dryrun_envvar_name.c_str(), std::to_string(nodes).c_str(), 1);
```

Comment:
For the device selection tests we've used a `config_testspy` instead of modifying the environment, imo that's a cleaner approach.

Reply:
I feel like using setenv in the unit tests is fine. It's slightly more integration- than unit-testy, but eh.

Reply:
not sure if we can do the same here, because in `device_selection_tests` the `config_testspy` is modifying a hand-made config, not the config that is initialized inside the runtime, which is what one of the things we want to test here.

Reply:
Right, because it's setting `num_nodes` in the constructor. In any case, modifying the global environment from a test is always iffy. What happens when the test fails? You'd need at least a RAII wrapper to ensure that the environment is always restored.

But maybe the cleanest solution would be to change the runtime ctor, like so:
```diff
-	runtime::runtime(int* argc, char** argv[], device_or_selector user_device_or_selector) {
+	runtime::runtime(int* argc, char** argv[], config cfg = {argc, argv}, device_or_selector user_device_or_selector) {
```